### PR TITLE
 Implement depth-first recursion in `wash find` + path primary

### PR DESCRIPTION
**Pull request by ekinanp**
Contributions to this project require sign-off consistent with the [Developers Certificate of Origin](https://developercertificate.org). This can be as simple as using `git commit -s` on each commit.

### CODE REVIEW

File: cmd/find/entry.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package cmdfind
+
+import (
+	"github.com/puppetlabs/wash/api/client"
+	apitypes "github.com/puppetlabs/wash/api/types"
+)
+
+type entry struct {
+	*apitypes.Entry
```
Comment (MikaelSmith) [inline cmd/find/entry.go]: I'm curious why this is a pointer. Not clear it's used that way.

Comment (ekinanp) [reply cmd/find/entry.go]: Oh, so we avoid doing a bunch of copy-by-values, which I think may be bad for hundreds of thousands of entries. Can change it back if you prefer the non-pointer version.

Comment (MikaelSmith) [reply cmd/find/entry.go]: I don't think it's actually avoiding those copies in lots of cases, and it's a small object so I wouldn't worry about it in the first place.

Comment (ekinanp) [reply cmd/find/entry.go]: Updated.

**Pull request merged**